src/perform_step/adams_bashforth_moulton_perform_step.jl

```diff
@@ -1536,3 +1536,59 @@ end
     cache.ϕstar_nm1[i] .= ϕstar_n[i]
   end
 end
+
+
+# ABCN2
+
+function initialize!(integrator,cache::ABCN2ConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = 
+  integrator.f.f1(integrator.uprev,integrator.p,integrator.t) +
+  integrator.f.f2(integrator.uprev,integrator.p,integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::ABCN2ConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack k2,uf,κ,tol = cache
+  cnt = integrator.iter
+  f1 = integrator.f.f1
+  f2 = integrator.f.f2
+  if cnt == 1
+    halfdt = dt/2
+    k = f1(uprev + halfdt*integrator.fsalfirst, p, t+halfdt) + f2(uprev + halfdt*integrator.fsalfirst, p, t+halfdt)
+    u = uprev + dt*k  # Midpoint
+    cache.k2 = f2(uprev, p ,t)
+  else
+    # Explicit part
+    k1 = f2(uprev, p, t)
```

Comment:
using the `zprev = dt*f1(uprev, p, t)` and the fsallast you can get `k1` without another `f2` evaluation. You can assume that `f2` is the expensive part since it's the non-stiff part that isn't being iterated.

Reply:
that should be `fsalfirst`? `k1 = dt*fsalfirst - zprev`

Reply:
yes. And be careful with the `dt` by changing the later calculation and you're good.

Reply:
okay